JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/BackEnd.java

```diff
@@ -88,9 +86,36 @@ public Receive<Event> actor() {
                 .onMessage(EventGetMatchesForReviewReq.class, this::eventGetMatchesForReviewHandler)
                 .onMessage(EventPatchUnLinkReq.class, this::eventPatchUnLinkHandler)
                 .onMessage(EventNotificationRequestReq.class, this::eventNotificationRequestHandler)
+                .onMessage(EventSearchReq.class, this::eventSearchHandler)
                 .build();
     }
 
+
+    private int distance(String strength) {
+        HashMap<String, Integer > map = new HashMap < String, Integer > ();
+        map.put("Exact Match", 0);
+        map.put("Low Fuziness", 1);
+        map.put("Medium Fuzziness", 2);
+        map.put("High Fuzziness", 3);
+
+        return map.get(strength);
+    }
+    private Behavior < Event > eventSearchHandler(EventSearchReq request) {
+        libMPI.startTransaction();
+        HashMap<String, Integer > map = new HashMap < String, Integer > ();
+        HashMap<String, String > fields = new HashMap < String, String > ();
+
+        for (int i = 0; i < (request.search().parameters()).size(); i++) {
+            map.put(request.search().parameters().get(i).field(), distance(request.search().parameters().get(i).distance()));
+            fields.put(request.search().parameters().get(i).field(), request.search().parameters().get(i).value());
+        }
+        var recs = libMPI.search(fields.get("national_id"), fields.get("given_name"), fields.get("family_name"), map);
```

Comment:
Note that the fields depend on the implementation, so we need : 
- Check the parameters fields against the fields config and filter only the one we have.
- Pass the parameters as a List to the libMPI.search() function.